pkg/termination/termination_test.go

```diff
@@ -0,0 +1,355 @@
+/*
+Copyright The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package termination
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	machinev1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var emptyEvents = func(rw http.ResponseWriter, req *http.Request) {
+	rw.Write([]byte(`{"DocumentIncarnation":0,"Events":[]}`))
+}
+
+var _ = Describe("Handler Suite", func() {
+	var terminationServer *httptest.Server
+	var httpHandler http.Handler
+	var nodeName string
+	var stop chan struct{}
+	var errs chan error
+	var h *handler
+
+	BeforeEach(func() {
+		// Reset test vars
+		terminationServer = nil
+		httpHandler = nil
+		nodeName = "test-node"
+		httpHandler = newMockHTTPHandler(emptyEvents)
+
+		handlerInterface, err := NewHandler(klogr.New(), cfg, 100*time.Millisecond, "", nodeName)
+		Expect(err).ToNot(HaveOccurred())
+
+		h = handlerInterface.(*handler)
+
+		h.pollURL = nil
```

Comment:
Can't we add a note why we are doing this

Reply:
done